src/include/plan_node/abstract_join_plan_node.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <memory>
+#include <utility>
+#include "parser/expression/abstract_expression.h"
+#include "plan_node/abstract_plan_node.h"
+
+namespace terrier::plan_node {
+
+//===--------------------------------------------------------------------===//
+// Abstract Join Plan Node
+//===--------------------------------------------------------------------===//
+
+class AbstractJoinPlanNode : public AbstractPlanNode {
+ public:
+  AbstractJoinPlanNode(std::shared_ptr<OutputSchema> output_schema, LogicalJoinType join_type,
+                       parser::AbstractExpression *predicate)
+      : AbstractPlanNode(std::move(output_schema)), join_type_(join_type), predicate_(predicate) {}
+
+  common::hash_t Hash() const override;
```

Comment:
can probably get rid of `hash_t` unless you can think of a reason we need it.

Reply:
I figured we were using it as a clarity tool to make it more readable in the code.